core/debug.cc

```diff
@@ -371,21 +372,37 @@ static bool SkipSymbol(char *symbol) {
   _exit(EXIT_FAILURE);
 }
 
+[[noreturn]] static void abort_failure() {
+  abort();
+}
+
 [[ gnu::noinline, noreturn ]] void GoPanic() {
   if (oops_msg == "")
     oops_msg = DumpStack();
 
-  // Create a crash log file
-  try {
-    std::ofstream fp(P_tmpdir "/bessd_crash.log");
-    fp << oops_msg;
-    fp.close();
-  } catch (...) {
-    // Ignore any errors.
+  // Create a crash log file if we are not dropping a core file.
+  if (!FLAGS_core_dump) {
```

Comment:
I guess the new behavior here is to either create a crashlog file or exit with `abort()`. I think these two options are not really exclusive to each other, i.e., one may want to create a core dump yet still have a crashlog file as well. Maybe I am misunderstood but this `if` statement looks unnecessary.

Reply:
Well you are right that I am tying together the the control of the crashlog file generation with the control of the core file. I suppose my real desire is to eliminate the creation of the crashlog file because: 1. It is already logged with LOG(FATAL) so it is superfluous. 2. Unlike glog, I can not control where is it dumped. I don't want my logs dumped to /tmp but prefer to dump them to /var/log/. With glog, this is controlled with an environment variable. 3. The crashlog is redundant with the core file.

Anyway, I am open to suggestions. I can either:
1. Add another command line argument to control whether or not the crash log is generated.
2. Document the -core-dump argument to indicate that when a core is dumped a crashlog file will not be generated.

Reply:
I like the first option better. The argument could be a boolean or a string (path to the crashlog, empty string if disabled). Either is fine by me.

As you said the crashlog is quite superfluous. `bessctl` would check if the file exists and show its content when the grpc connection abruptly has been disconnected. I was too lazy to make it find and read the log file and show the last log messages, so created a new file.

Reply:
I will add a boolean argument. I did not realize the relationship with bessctl. Thanks for the explanation.